odoo/addons/base/ir/ir_model.py

```diff
@@ -912,10 +913,39 @@ def _add_manual_fields(self, model):
         fields_data = self._get_manual_field_data(model._name)
         for name, field_data in fields_data.items():
             if name not in model._fields and field_data['state'] == 'manual':
-                field = self._instanciate(field_data)
-                if field:
-                    model._add_field(name, field)
-
+                modules_in_addon_path = get_modules()
+                custom_modules = [r['name'] for r in self.env['ir.module.module'].search_read([
+                ('name', 'not in', modules_in_addon_path + ['studio_customization'])
+                ], ['name'])]
+                try:
+                    field = self._instanciate(field_data)
+                    if field:
+                        model._add_field(name, field)
+                except:
+                    self.env.cr.execute(
+                        """
+                        SELECT d.module
+                        FROM ir_model m
+                        JOIN ir_model_data d ON d.res_id = m.id AND d.model = 'ir.model'
+                        WHERE m.model = %s
+                        """,
+                        [field_data["relation"]],
```

Comment:
`relation` is not the only field we should consider to determine if the field relies on model/fields not in the addons path.
`related` can have the case as well. Have a look at `related_field_id`, `relation_field_id` and `relation_field`. Maybe they are not all necessary, but some of them could.

Reply:
Add some code in another place